backend/api/views.py

```diff
@@ -136,16 +136,19 @@ def accept_scrimmage_helper(scrimmage_id):
     blue_team_id = scrimmage.blue_team.id
     red_submission_id = scrimmage.red_submission_id
     blue_submission_id = scrimmage.blue_submission_id
+    # save the scrimmage, again, to mark save
+    if red_submission_id is None or blue_submission_id is None:
+        scrimmage.status = 'error'
+        scrimmage.error_msg = 'Make sure your team and the team you requested have a submission.'
+    else:
+        scrimmage.status = 'queued'
+    scrimmage.save()
     red_team_name = Team.objects.get(pk=red_team_id).name
     blue_team_name = Team.objects.get(pk=blue_team_id).name
     replay = scrimmage.replay
     map_ids = scrimmage.map_ids
     scrimmage_pub_sub_call(red_submission_id, blue_submission_id, red_team_name, blue_team_name, scrimmage.id, replay, map_ids)
 
-    # save the scrimmage, again, to mark save
```

Comment:
imo should _also_ save here

Reply:
actually if you want to avoid a double save:

if error:
    scrimmage.status = 'error'
    scrimmage.save()
    return Response(...)
(and then keep going)

Reply:
as in move it to after the pubsub call?

Reply:
in case of error, is better to save and quit before the pubsub call. (so that we don't put some invalid requests on the pubsub)
otherwise, we want to call the pubsub and then save, as before

Reply:
wait but doesn't the pubsub handle cases (returns 400, doesnt actually pubsub) in which there is an error? so its fine to just call it after right?

Reply:
ohhh you right, i just noticed -- nice catch

Reply:
looks good then, thanks for clarifying!
in the long run I feel like we should refactor scrimmage_pub_sub_call to be a more self-contained method. (eg like these checks to me make more sense in the method that creates scrims, rather than the method that just focuses on the pubsub). but certainly for later -- is just a design decision.